app/models/question.rb

```diff
@@ -18,8 +20,12 @@ class Question < ActiveRecord::Base
   # No questions in one application template with the same number
   validates :number, uniqueness: { scope: :application_template,
                                    allow_blank: true }
+  validates :number,
+            uniqueness: { scope: :application_draft,
+                          if: -> { application_draft.present? } }
 
   default_scope { order :number }
+  scope :not_new, -> { where.not id: nil }
```

Comment:
I'd be open to suggestions of a better AR scope name here, I don't like `not_new`.